---
title: "RSV_analysis"
author: "Nitsueh Kebere and Jessica McClintock"
date: "5/10/2022"
output: html_document
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r, message=FALSE, warning=FALSE}
setwd("/Users/jessmcc/Documents/workProjects/ZPRIME_RSV")
library(tidyverse)
library(devtools) 
library(phyloseq) 
library(ggpubr)
library(paletteer)
```

# Read in Data

```{r message=FALSE, warning=FALSE}

otu_mat <- read_csv("data/1pct/Assay_count_1pct.csv")
tax_mat <- read_csv("data/1pct/Assay_taxonomy_1pct.csv")
samples_df <- read_csv("data/1pct/Assay_annotation_1pct.csv")

```

## Create Phyloseq Object
All analysis was conducted based on a phyloseq object. 
Changed metadata labels to be more descriptive for table and graphing 
purposes.

```{r message=FALSE, warning=FALSE}
OTU <- otu_mat %>%
  tibble::column_to_rownames("...1") %>% 
  as.matrix() %>%
  otu_table(taxa_are_rows = TRUE)

TAX <- tax_mat %>% 
  tibble::column_to_rownames("...1") %>% 
  as.matrix() %>%
  tax_table()

samples <- samples_df %>% 
  tibble::column_to_rownames("...1") %>%
  sample_data()

samples$state[samples$state == "pos"] <- "RSV+"
samples$state[samples$state == "neg"] <- "RSV-"

samples$mother_hiv[samples$mother_hiv == 0] <- "Unexposed"
samples$mother_hiv[samples$mother_hiv == 1] <- "Exposed"

colnames(samples)[5] <- "sex"
samples$sex[samples$sex == 1] <- "Female"
samples$sex[samples$sex == 2] <- "Male"
samples$sex[samples$sex == 8] <- "Unknown"

for(sample in 1:length(samples$death_age)){
    if(samples$death_age[sample] < 29){
        samples$death_age[sample] <- 0
    }else if(samples$death_age[sample] > 28 & samples$death_age[sample] < 62){
        samples$death_age[sample] <- 1
    }else if(samples$death_age[sample] > 61 & samples$death_age[sample] < 153){
        samples$death_age[sample] <- 2
    }else{
        samples$death_age[sample] <- 4
    }
}
samples$death_age <- as.factor(samples$death_age)
zprime <- phyloseq(OTU, TAX, samples)

```

# Chi-Squared test comparing demographic information
Test to see if there is a difference in the demographic make up of the HIV+
group compared to the HIV- group using chi-square goodness of fit since this is qualitative data.

```{r message=FALSE, warning=FALSE}
genderCounts <- table(samples_df$state, samples_df$gender)

chi_stats <- list()

chi_stats$gender <- chisq.test(samples_df$state, samples_df$gender)$p.value
chi_stats$hiv <-chisq.test(samples_df$state, samples_df$mother_hiv)$p.value
chi_stats$death_age <- chisq.test(samples_df$state, samples_df$death_age)$p.value
chi_stats$location <- chisq.test(samples_df$state, samples_df$location)$p.value
chi_stats$hospitalized <- chisq.test(samples_df$state, samples_df$hosp_duration)$p.value
print(chi_stats)
```
# Create color palette that represents 15 organisms well
Selected color palette for all graphing
```{r message=FALSE, warning=FALSE}
chosen_pal_16 <- "awtools::bpalette"
chosen_pal_5 <- "fishualize::Scarus_hoefleri"

usepalette <- paletteer::paletteer_d(chosen_pal_16, direction = -1)

usepalette <- c(paletteer::palettes_d[["Polychrome"]]$kelly)

usepalette <- c(usepalette[which(usepalette != "#f2f3f4")]) 
usepalette <- c(usepalette[which(usepalette != "#222222")])
usepalette <- c(usepalette[which(usepalette != "#f3c300")])

usepalette[16] <-"#f2f3f4"

```

# RSV Status
## Determine Relative Abundance

```{r message=FALSE, warning=FALSE}
relativeAbundance <- function(x){
  x / sum(x)
}

zp_rel_abund <- transform_sample_counts(zprime, relativeAbundance)

```

### Rank Most Abundant Genera 

```{r messafe=FALSE, warning=FALSE}

rankedAbundancePos <- psmelt(zp_rel_abund) %>%
    group_by(state, OTU) %>%
    summarise(rel_abund = mean(Abundance)) %>%
    filter(state == "RSV+") %>%
    arrange(desc(rel_abund))

#Create levels in order of abundance, but keep Order at the top of the list so it will plot at the top of the chart; used in next code chunk after phyloseq melt

level_for_factor <- c(rankedAbundancePos$OTU[rankedAbundancePos$OTU != "Other"], "Other")

rankedAbundanceNeg <- psmelt(zp_rel_abund) %>%
    group_by(state, OTU) %>%
    summarise(rel_abund = mean(Abundance)) %>%
    filter(state == "RSV-") %>%
    arrange(desc(rel_abund))

```
### Visualizing Relative Abundance 
Relative abundances show the microbes present in each samples' microbiome as a percentage of the whole microbiome. Percentages will differ from sample to sample even if actual counts of specific microbes are similar. Or, percentages may seem similar when the counts are actually VERY different.

```{r message=FALSE, warning=FALSE}

zp_rel_abund_df <- psmelt(zp_rel_abund)

zp_rel_abund_df$genus <- factor(zp_rel_abund_df$genus, levels = level_for_factor)

ggplot(zp_rel_abund_df, aes(color = genus, fill = genus, y=Abundance, x=Sample)) +
  geom_bar(stat = "identity", position = "stack") +
  labs(x = "Sample", y = "Relative Abundance\n") +
  facet_wrap(~state, scales = "free") +
  theme(panel.background = element_blank(),
        axis.text.x=element_blank(),
        axis.ticks.x=element_blank(),
        legend.key.size = unit(0.2, "cm")) +
  scale_fill_manual(values = usepalette) +
  scale_color_manual(values = usepalette)

ggsave("Plots/1pct/relativeAbundance.jpeg")
```

## Testing for a difference in genus level abundance 
Analogous to a two-sample t-test, but instead we are testing whether taxa 
frequencies observed in both groups of metagenomic samples are equal. A 
Dirichlet-Multinomial distribution is assumed for the data.  The HMP 
test null hypothesis is that there is no difference in the distribution 
of phyla between groups. p-values > 0.05 support the null hypothesis.

```{r, message=FALSE, warning=FALSE}
zp_genus <- tax_glom(zp_rel_abund, taxrank = "genus") #relative abundance
zp_genus_counts <- tax_glom(zprime, taxrank = "genus") #absolute abundance by genus

zp_pos <- subset_samples(zp_genus_counts, state == "RSV+")
zp_neg <- subset_samples(zp_genus_counts, state =="RSV-")

zp_pos_otu <- otu_table(zp_pos) %>%
  data.frame() %>%
  t() %>%
  as.data.frame()

zp_neg_otu <- otu_table(zp_neg) %>%
  data.frame() %>%
  t() %>%
  as.data.frame()

group_data <- list(zp_pos_otu, zp_neg_otu)
(xdc <- HMP::Xdc.sevsample(group_data)) 
```

## Alpha Diversity 
Alpha diversity quantifies diversity within individual samples and compares the differences across groups. These analysis aim to determine the number of species in a given sample (richness) and compare averages in one group (RSV+) to another group (RSV-). Some methods will also compare evenness, or how the abundance of each individual species in a sample compares to the abundance of other species (frequency). The Shannon index combines richness and frequency. 

### Shannon and Simpson Indexes
Comparing the total number of observed genera.
#### Visualization
```{r, message=FALSE, warning=FALSE}
adiv <- data.frame(
  "Observed" = estimate_richness(zprime, measures = "Observed"),
  "Shannon" = estimate_richness(zprime, measures = "Shannon"),
  "Simpson" = estimate_richness(zprime, measures = "Simpson"),
  "Status" = sample_data(zprime)$state)

adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Status, y = value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = Status), height = 0, width = .2) +
  ggtitle("Genera") +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none", plot.title = element_text(hjust = 0.5))+
  scale_color_hue(l=45)

ggsave("Plots/1pct/shannonAndSimpson/state.jpeg")
```

#### Statistical Testing
```{r, message=FALSE, warning=FALSE}
adiv %>%
  group_by(Status) %>%
  summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_simpson = median(Simpson))

# Test for normality
shapiro.test(subset(adiv, Status == "RSV+")$Shannon)
shapiro.test(subset(adiv, Status == "RSV-")$Shannon)
shapiro.test(subset(adiv, Status == "RSV+")$Simpson)
shapiro.test(subset(adiv, Status == "RSV-")$Simpson)

# pos samples are not normally distributed, neg samples are, so use 
wilcox.test(Observed ~ Status, data = adiv, exact = FALSE, conf.int = TRUE)
wilcox.test(Shannon ~ Status, data = adiv, conf.int = TRUE) 
wilcox.test(Simpson ~ Status, data = adiv, conf.int = TRUE) 

```
## Beta Diversity 
Beta Diversity aims to analyze feature dissimilarity between each pair of samples and utilize distance analysis, like Bray-Curtis or UniFrac. Quantitivate methods, like Bray-Curtis or weighted UniFrac, look at abundance of features in determining distances whereas Qualitative measures, like unweighted uniFrac or binary Jaccard, only look at presence or absence of features. PERMANOVA is used to determine clustering between groups.These are then visualized via PCA or PCoA

### HeatMap 
Visualize the Bray-curtis dissimilarity (as two samples share fewer taxa, the number increases 0 for exact same composition, 1 for no shared genera) results.

```{r, message=FALSE, warning=FALSE, fig.width=10}
zp_scale <- zp_rel_abund %>%
    otu_table() %>%
    t() %>%
    scale() %>%
    t() %>%
    otu_table(taxa_are_rows = TRUE)

zpscale <- zp_rel_abund

otu_table(zpscale) <- zp_scale

p <- plot_heatmap(zpscale, method = "NMDS", distance = "bray",
             sample.label = "state", sample.order = "state")
axis_color <- ifelse(p$scales$scales[[1]]$labels == "RSV+", "red", "blue")
p <- p + theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 10, colour = axis_color), plot.margin = unit(c(1,1,1,1),"cm"))

ggsave("Plots/1pct/heatmap.jpeg")
```

### PCoA

```{r, message=FALSE, warning=FALSE}
#Atchinson Distance PCoA
clr_dist_matrix <- phyloseq::distance(zp_rel_abund, method = "euclidean") 
eucDis <- ordinate(zp_rel_abund, method = "PCoA", distance = clr_dist_matrix )
PCoA_Atchison_plot <- plot_ordination(zp_rel_abund, eucDis, color = "state")

ggsave("Plots/1pct/PCoA_Atchinson.jpeg")

#Bray-Curtis Distance PCoA
pcoa <- ordinate(zp_rel_abund, method = "PCoA", distance = "bray", weighted = TRUE)

PCoA_Bray_plot <- plot_ordination(zp_rel_abund, pcoa, color = "state") +
    scale_color_hue(l=45) + 
    stat_ellipse(lineype = 2)

ggsave("Plots/1pct/PCoA_Bray.jpeg")

```

### NMDS
```{r, message=FALSE, warning=FALSE}

zprime.ord <- ordinate(zp_rel_abund, "NMDS", "bray")

NMDS_plot <- plot_ordination(zp_rel_abund, zprime.ord, type="samples", color="state") + 
    #geom_point(size=3) +
    scale_color_hue(l=45) +
    stat_ellipse(linetype = 2)


ggsave("Plots/1pct/NMDS.jpeg")
```

### Create one Bray-Curtis Figure
```{r message=FALSE, warning=FALSE, fig.width=10}
brayCurtisFigure <- ggarrange(p, 
                              ggarrange(NMDS_plot, PCoA_Bray_plot, 
                                        ncol = 2, labels = c("B", "C"),
                                        common.legend = TRUE), 
                    labels = c("A"),nrow = 2, heights = c(2,1.5))+
    bgcolor("white")
brayCurtisFigure

ggsave("Plots/1pct/BrayCurtisFigure.jpeg")
```

## Differential Abundance Testing
### Wilcoxon-rank sum to determine significant difference between any specific genus
Perform a  non-parametric Wilcoxon rank-sum test where the null hypothesis is equal medians between states. 

```{r message=FALSE, warning=FALSE, fig.width=10}
wilcox_test <- list()

zp_pos <- subset_samples(zp_rel_abund, state == "RSV+")
zp_neg <- subset_samples(zp_rel_abund, state =="RSV-")

zp_pos_otu <- otu_table(zp_pos) %>%
  data.frame() %>%
  t() %>%
  as.data.frame()

zp_neg_otu<- otu_table(zp_neg)%>%
  data.frame() %>%
  t() %>%
  as.data.frame()

for(genus in colnames(zp_pos_otu)){
    wilcox_test[[genus]] <- round(wilcox.test(zp_pos_otu[[genus]], zp_neg_otu[[genus]])$p.value, 4)
}

wilcox_test <- as.data.frame(wilcox_test, row.names = "RSV_state") %>%
    t()

wilcox_test <- wilcox_test %>%
   as.data.frame() %>%
   mutate(BH_FDR_state = round(p.adjust(RSV_state, "BH"), 4))

print(wilcox_test)
```
#### Visulaizing Significantly Significant Abundance Grouped by RSV state
Allows visualization of each microbe that has a wilcoxon rankn sum p value <0.05.

```{r message=FALSE, warning=FALSE, fig.width=5}

significant_taxa <-  wilcox_test %>%
    as.data.frame() %>%
    filter(RSV_state < 0.05)

if(length(significant_taxa$RSV_state) != 0){
 
print(psmelt(zp_rel_abund) %>%
  filter(genus %in% row.names(significant_taxa)) %>%
  ggplot(data = ., aes(x = state, y = Abundance)) +
  geom_boxplot(outlier.shape  = NA) +
  geom_jitter(aes(color = genus), height = 0, width = .2) +  
  labs(title = "Relative Abundance based on RSV Status", x = "", y = "Relative Abundance\n") +
  facet_wrap(~ OTU, scales = "free") +
  theme(legend.position = "none") +
  scale_color_hue(l=45))


ggsave("Plots/1pct/wilcox_test_sig_state.jpeg", dpi = 300, width = 5, height = 4)

}else{
    print("No signifiant taxa via wilcoxon test for RSV status.")
}
```

# Location
## Testing for a difference in genus level abundance by location 

```{r, message=FALSE, warning=FALSE}
zp_BID <- subset_samples(zp_genus_counts, location == "BID")
zp_Hospital <- subset_samples(zp_genus_counts, location == "Hospital")

zp_BID_otu <- otu_table(zp_BID) %>%
  data.frame() %>%
  t()%>%
  as.data.frame()

zp_Hospital_otu <- otu_table(zp_Hospital) %>%
  data.frame() %>%
  t()%>%
  as.data.frame()

group_data <- list(zp_BID_otu, zp_Hospital_otu)
(xdc_location <- HMP::Xdc.sevsample(group_data)) 
```
## Alpha Diversity 

### Shannon and Simpson Indexes
#### Visualization
```{r, message=FALSE, warning=FALSE}
adiv <- data.frame(
  "Observed" = estimate_richness(zprime, measures = "Observed"),
  "Shannon" = estimate_richness(zprime, measures = "Shannon"),
  "Simpson" = estimate_richness(zprime, measures = "Simpson"),
  "Status" = sample_data(zprime)$location)

adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Status, y = value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = Status), height = 0, width = .2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none") +
  scale_color_hue(l=45)

ggsave("Plots/1pct/shannonAndSimpson/location.jpeg")
```
#### Statistical Test
```{r, message=FALSE, warning=FALSE}
adiv %>%
  group_by(Status) %>%
  summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_simpson = median(Simpson))

# Test for normality
shapiro.test(subset(adiv, Status == "BID")$Shannon)
shapiro.test(subset(adiv, Status == "Hospital")$Shannon)
shapiro.test(subset(adiv, Status == "BID")$Simpson)
shapiro.test(subset(adiv, Status == "Hospital")$Simpson)

# Only BID Simpson are normally distributed, so use wilcoxon
wilcox.test(Observed ~ Status, data = adiv, exact = FALSE, conf.int = TRUE)
wilcox.test(Shannon ~ Status, data = adiv, conf.int = TRUE) 
wilcox.test(Simpson ~ Status, data = adiv, conf.int = TRUE) 

```

## Differential Abundance Testing
### Wilcoxon rank-sum for significant difference between location
Perform wilcoxon rank sum to determine if any genus differs significantly
between early facility deaths or community deaths.
```{r message=FALSE, warning=FALSE, fig.width=10}
zp_clr_BID <- subset_samples(zp_rel_abund, location =="BID")
zp_clr_hospital <- subset_samples(zp_rel_abund, location == "Hospital")

zp_clr_BID_otu <- otu_table(zp_clr_BID) %>%
  data.frame() %>%
  t() %>%
  as.data.frame()

zp_clr_hospital_otu <- otu_table(zp_clr_hospital) %>%
  data.frame() %>%
  t() %>%
  as.data.frame()

p_vals <- list()

for(genus in colnames(zp_clr_BID_otu)){
    p_vals[[genus]] <-round(wilcox.test(zp_clr_BID_otu[[genus]], zp_clr_hospital_otu[[genus]])$p.value, 4)
}

p_vals <- as.data.frame(p_vals, row.names = "location") %>%
    t()

wilcox_test <- cbind(wilcox_test, p_vals)

wilcox_test <- wilcox_test %>%
   as.data.frame() %>%
   mutate(BH_FDR_location = round(p.adjust(location, "BH"), 4))


print(wilcox_test)
```

#### Visualize signigficant taxa by location
```{r message=FALSE, warning=FALSE}
significant_taxa <-  wilcox_test %>%
    as.data.frame() %>%
    filter(location < 0.05)

if(length(significant_taxa$location) != 0){
    locationFig <- psmelt(zp_rel_abund) %>%
              filter(genus %in% row.names(significant_taxa)) %>%
              ggplot(data = ., aes(x = location, y = Abundance)) +
              geom_boxplot(outlier.shape  = NA) +
              geom_jitter(aes(color = genus), height = 0, width = .2) +
              labs(title = "Relative Abundance Based on Location", x = "", y = "Abundance\n") +
              facet_wrap(~ OTU, scales = "free") +
              theme(legend.position = "none") +
              scale_color_hue(l=45) #, plot.title=element_text(hjust=0.5, size=18))
    
    locationFig
    ggsave("Plots/1pct/wilcox_test_sig_location.jpeg")
}else{
    print("No signifiant taxa via wilcoxon test for Location.")
}

```

# HIV Exposure
## Testing for a difference in genus level abundance by HIV Exposure Status

```{r, message=FALSE, warning=FALSE}
zp_HIV_neg <- subset_samples(zp_genus_counts, mother_hiv == "Unexposed")
zp_HIV_pos <- subset_samples(zp_genus_counts, mother_hiv == "Exposed")

zp_HIV_neg_otu <- otu_table(zp_HIV_neg) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_HIV_pos_otu <- otu_table(zp_HIV_pos) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

group_data <- list(zp_HIV_neg_otu, zp_HIV_pos_otu)
(xdc_location <- HMP::Xdc.sevsample(group_data)) 
```

## Alpha Diversity 
### Shannon and Simpson Indexes
#### Visualization
```{r, message=FALSE, warning=FALSE}
adiv <- data.frame(
  "Observed" = estimate_richness(zprime, measures = "Observed"),
  "Shannon" = estimate_richness(zprime, measures = "Shannon"),
  "Simpson" = estimate_richness(zprime, measures = "Simpson"),
  "Status" = sample_data(zprime)$mother_hiv)

adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Status, y = value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = Status), height = 0, width = .2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none") +
  scale_color_hue(l=45)

ggsave("Plots/1pct/shannonAndSimpson/mother_hiv.jpeg")
```
#### Statistical Test
```{r, message=FALSE, warning=FALSE}
adiv %>%
  group_by(Status) %>%
  summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_simpson = median(Simpson))

# Test for normality
shapiro.test(subset(adiv, Status == "Unexposed")$Shannon)
shapiro.test(subset(adiv, Status == "Exposed")$Shannon)
shapiro.test(subset(adiv, Status == "Unexposed")$Simpson)
shapiro.test(subset(adiv, Status == "Exposed")$Simpson)

# Only Unxposed Simpson are normally distributed, so use wilcoxon test
wilcox.test(Observed ~ Status, data = adiv, exact = FALSE, conf.int = TRUE)
wilcox.test(Shannon ~ Status, data = adiv, conf.int = TRUE) 
wilcox.test(Simpson ~ Status, data = adiv, conf.int = TRUE) 

```

## Differential Abundance Testing
### Wilcoxon rank sum test for significant difference between HIV status
Perform wilcoxon rank sum to determine if any genus differs significantly
between mother's HIV status.
```{r message=FALSE, warning=FALSE, fig.width=10}
zp_HIV_neg <- subset_samples(zp_rel_abund, mother_hiv == "Unexposed")
zp_HIV_pos <- subset_samples(zp_rel_abund, mother_hiv == "Exposed")

zp_HIV_neg_otu <- otu_table(zp_HIV_neg) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_HIV_pos_otu <- otu_table(zp_HIV_pos) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

p_vals <- list()

for(genus in colnames(zp_HIV_neg_otu)){
    p_vals[[genus]] <- round(wilcox.test(zp_HIV_neg_otu[[genus]], zp_HIV_pos_otu[[genus]])$p.value, 4)
}

p_vals <- as.data.frame(p_vals, row.names = "HIV_status") %>%
    t()
wilcox_test <- cbind(wilcox_test, p_vals)

wilcox_test <- wilcox_test %>%
   as.data.frame() %>%
   mutate(BH_FDR_HIV = round(p.adjust(HIV_status, "BH"), 4))

print(wilcox_test)
```

#### Plot signigficant taxa by HIV exposure
```{r message=FALSE, warning=FALSE}
significant_taxa <-  wilcox_test %>%
    as.data.frame() %>%
    filter(HIV_status < 0.05)

if(length(row.names(significant_taxa)) != 0){
    print(psmelt(zp_rel_abund) %>%
        filter(genus %in% row.names(significant_taxa)) %>%
        ggplot(data = ., aes(x = as.factor(mother_hiv), y = Abundance)) +
        geom_boxplot(outlier.shape  = NA) +
        geom_jitter(aes(color = genus), height = 0, width = .2) +
        labs(title = "Relative Abundance Based on Mother's HIV Status", 
             x = "", y = "Abundance\n") +
        facet_wrap(~ OTU, scales = "free") +
        theme(legend.position = "none", plot.title=element_text(hjust=0.5, size=18)) +
        scale_color_hue(l=45))
    
    ggsave("Plots/1pct/wilcox_test_sig_mother_hiv.jpeg")
}else{
    print("No significant taxa based on HIV exposure")
}
```

# Sex
## Testing for a difference in genus level abundance by Sex
```{r, message=FALSE, warning=FALSE}
zp_female <- subset_samples(zp_genus_counts, sex == "Female")
zp_male <- subset_samples(zp_genus_counts, sex == "Male")
zp_unknown <- subset_samples(zp_genus_counts, sex == "Unknown")

zp_female_otu <- otu_table(zp_female) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_male_otu <- otu_table(zp_male) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_unknown_otu <- otu_table(zp_unknown) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

group_data <- list(zp_female_otu, zp_male_otu, zp_unknown_otu)
(xdc_location <- HMP::Xdc.sevsample(group_data)) 
```

## Alpha Diversity 
### Shannon and Simpson Indexes
#### Visualization
```{r, message=FALSE, warning=FALSE}
adiv <- data.frame(
  "Observed" = estimate_richness(zprime, measures = "Observed"),
  "Shannon" = estimate_richness(zprime, measures = "Shannon"),
  "Simpson" = estimate_richness(zprime, measures = "Simpson"),
  "Status" = sample_data(zprime)$sex)

adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Status, y = value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = Status), height = 0, width = .2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none") +
  scale_color_hue(l=45)

ggsave("Plots/1pct/shannonAndSimpson/mother_hiv.jpeg")
```
#### Statistical Test
```{r, message=FALSE, warning=FALSE}
adiv %>%
  filter(Status != "Unknown")  %>%
  group_by(Status) %>%
  summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_simpson = median(Simpson))

adiv <- filter(adiv, Status != "Unknown")

# Test for normality
shapiro.test(subset(adiv, Status == "Male")$Shannon)
shapiro.test(subset(adiv, Status == "Female")$Shannon)
shapiro.test(subset(adiv, Status == "Male")$Simpson)
shapiro.test(subset(adiv, Status == "Female")$Simpson)

#Only simpson data is normally distributed, so use wilcox 
wilcox.test(Observed ~ Status, data = adiv, exact = FALSE, conf.int = TRUE)
wilcox.test(Shannon ~ Status, data = adiv, conf.int = TRUE) 
wilcox.test(Simpson ~ Status, data = adiv, conf.int = TRUE) 

```

## Differential Abundance Testing
### Wilcox for significant difference between Sex for any specific genus
Perform two sample t-test to determine if any genus differs significantly
between early facility deaths or community deaths.

```{r message=FALSE, warning=FALSE, fig.width=10}
zp_female <- subset_samples(zp_rel_abund, sex == "Female")
zp_male <- subset_samples(zp_rel_abund, sex == "Male")
zp_unknown <- subset_samples(zp_rel_abund, sex == "Unknown")

zp_female_otu <- otu_table(zp_female) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_male_otu <- otu_table(zp_male) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_unknown_otu <- otu_table(zp_unknown) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

microbes <- colnames(zp_female_otu)
sex <- "Female"
zp_female_otu <- cbind(zp_female_otu, sex)
sex <- "Male"
zp_male_otu <- cbind(zp_male_otu, sex)

all_data <- rbind(zp_female_otu, zp_male_otu)

p_vals <- list()

for(genus in microbes){
    p_vals[[genus]] <- round(wilcox.test(zp_female_otu[[genus]], zp_male_otu[[genus]])$p.value, 4)
}

p_vals <- as.data.frame(p_vals, row.names = "sex") %>%
    t() %>%
    as.data.frame() 

wilcox_test <- cbind(wilcox_test, p_vals)

wilcox_test <- wilcox_test %>%
   as.data.frame() %>%
   mutate(BH_FDR_sex = round(p.adjust(sex, "BH"), 4))

print(wilcox_test)
```

#### Plot signigficant taxa by Sex
```{r message=FALSE, warning=FALSE}
significant_taxa <-  wilcox_test %>%
    as.data.frame() %>%
    filter(sex < 0.05)

if(length(row.names(significant_taxa))> 0){
  sexFig <- psmelt(zp_rel_abund) %>%
  filter(genus %in% row.names(significant_taxa)) %>%
  filter(sex %in% c("Female", "Male")) %>%
  ggplot(data = ., aes(x = sex, y = Abundance)) +
  geom_boxplot(outlier.shape  = NA) +
  geom_jitter(aes(color = genus), height = 0, width = .2) +
  labs(title = "Relative Abundance Based on Sex", x = "", y = "Abundance\n") +
  facet_wrap(~ OTU, scales = "free") +
  theme(legend.position = "none", plot.title=element_text(hjust=0.5, size=18)) +
  scale_color_hue(l=45)
  
  sexFig
  
  ggsave("Plots/1pct/wilcox_sig_sex.jpeg")
}else{
    print("No significance based on sex.")
}

```

# Age
## Testing for a difference in genus level abundance by location 

```{r, message=FALSE, warning=FALSE}
zp_newborn <- subset_samples(zp_genus_counts, death_age == 0)
zp_one_month <- subset_samples(zp_genus_counts, death_age == 1)
zp_two_month <- subset_samples(zp_genus_counts, death_age == 2)
zp_five_month <- subset_samples(zp_genus_counts, death_age == 4)

zp_newborn_otu <- otu_table(zp_newborn) %>%
  data.frame() %>%
  t()%>%
  as.data.frame()

zp_one_month_otu <- otu_table(zp_one_month) %>%
  data.frame() %>%
  t()%>%
  as.data.frame()

zp_two_month_otu <- otu_table(zp_two_month) %>%
  data.frame() %>%
  t()%>%
  as.data.frame()

zp_five_month_otu <- otu_table(zp_five_month) %>%
  data.frame() %>%
  t()%>%
  as.data.frame()

group_data <- list(zp_newborn_otu, zp_one_month_otu, zp_two_month_otu, zp_five_month_otu)
(xdc_location <- HMP::Xdc.sevsample(group_data)) 
```

## Alpha Diversity 
### Shannon and Simpson Indexes
Comparing the total number of observed genera.

```{r, message=FALSE, warning=FALSE}
adiv <- data.frame(
  "Observed" = estimate_richness(zp_genus_counts, measures = "Observed"),
  "Shannon" = estimate_richness(zp_genus_counts, measures = "Shannon"),
  "Simpson" = estimate_richness(zp_genus_counts, measures = "Simpson"),
  "Status" = sample_data(zp_genus_counts)$death_age)

adiv %>%
  gather(key = metric, value = value, c("Observed", "Shannon", "Simpson")) %>%
  mutate(metric = factor(metric, levels = c("Observed", "Shannon", "Simpson"))) %>%
  ggplot(aes(x = Status, y = value)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(aes(color = Status), height = 0, width = .2) +
  labs(x = "", y = "") +
  facet_wrap(~ metric, scales = "free") +
  theme(legend.position="none") +
  scale_color_hue(l=45)

ggsave("Plots/1pct/shannonAndSimpson/age.jpeg")
```

#### Statistical Test
```{r, message=FALSE, warning=FALSE}
adiv %>%
  group_by(Status) %>%
  summarise(median_observed = median(Observed),
            median_shannon = median(Shannon),
            median_simpson = median(Simpson))

# Test for normality
shapiro.test(subset(adiv, Status == 0)$Shannon)
shapiro.test(subset(adiv, Status == 1)$Shannon)
shapiro.test(subset(adiv, Status == 2)$Shannon)
shapiro.test(subset(adiv, Status == 4)$Shannon)
shapiro.test(subset(adiv, Status == 0)$Simpson)
shapiro.test(subset(adiv, Status == 1)$Simpson)
shapiro.test(subset(adiv, Status == 2)$Simpson)
shapiro.test(subset(adiv, Status == 4)$Simpson)

#Only age group 0, 1 and 2 in the Simpson are normally distributed, so use Kruskal-willis
kruskal.test(Observed ~ Status, data = adiv)
kruskal.test(Shannon ~ Status, data = adiv) 
kruskal.test(Simpson ~ Status, data = adiv) 

```
## Differential Abundance Testing
### Kruskal-Wallis for significant difference between Age for any specific genus
Perform non parametric Kruskal-Wallis to determine if any genus differs significantly
between age of death.

```{r message=FALSE, warning=FALSE, fig.width=10}
zp_zero <- subset_samples(zp_rel_abund, death_age  == 0)
zp_one <- subset_samples(zp_rel_abund, death_age == 1)
zp_two <- subset_samples(zp_rel_abund, death_age == 2)
zp_four <- subset_samples(zp_rel_abund, death_age == 4)

zp_zero_otu <- otu_table(zp_zero) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_one_otu <- otu_table(zp_one) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_two_otu <- otu_table(zp_two) %>%
  data.frame() %>%
  t() %>%
  as.data.frame() 

zp_four_otu <- otu_table(zp_four) %>%
   data.frame() %>%
   t() %>%
   as.data.frame() 

microbes <- colnames(zp_zero_otu)
age <- "0"
zp_zero_otu <- cbind(zp_zero_otu, age)
age <- "1"
zp_one_otu <- cbind(zp_one_otu, age)
age <- "2"
zp_two_otu <- cbind(zp_two_otu, age)
age <- "4"
zp_four_otu <- cbind(zp_four_otu, age)

all_data <- data.frame()
all_data <- rbind(zp_zero_otu, zp_one_otu, zp_two_otu, zp_four_otu)

p_vals <- list()

for(genus in microbes){
    p_vals[[genus]] <- round(kruskal.test(all_data[[genus]] ~ all_data$age)$p.value, 4)
}

p_vals <- as.data.frame(p_vals, row.names = "age") %>%
    t() %>%
    as.data.frame() 

wilcox_test <- cbind(wilcox_test, p_vals)

wilcox_test <- wilcox_test %>%
   as.data.frame() %>%
   mutate(BH_FDR_age = round(p.adjust(age, "BH"), 4))

print(wilcox_test)
```

#### Plot signigficant taxa by age
```{r message=FALSE, warning=FALSE}
significant_taxa <-  wilcox_test %>%
    as.data.frame() %>%
    filter(age < 0.05)

if(length(row.names(significant_taxa)) != 0){
  ageFig <-  psmelt(zp_rel_abund) %>%
  filter(genus %in% row.names(significant_taxa)) %>%
  ggplot(data = ., aes(x = as.factor(death_age), y = Abundance)) +
  geom_boxplot(outlier.shape  = NA) +
  geom_jitter(aes(color = genus), height = 0, width = .2) +
  labs(title = "Relative Abundance Based on Age", x = "", y = "Abundance\n") +
  facet_wrap(~ OTU, scales = "free") +
  theme(legend.position = "none", plot.title=element_text(hjust=0.5, size=18)) +
  scale_color_hue(l=45)

  ageFig
  
  ggsave("Plots/1pct/kruskal_Age.jpeg")
}else{
    print("No significant difference based on length of hospital stay.")
}

```

#Clean p-value chart
```{r, message=FALSE, warning=FALSE}
#wilcox_test <- dplyr::select(wilcox_test, -len_hosp_stay, -BH_FDR_len_hosp_stay) 

for(col in 1:ncol(wilcox_test)){
    for(item in 1:length(wilcox_test[,col])){
        if (wilcox_test[item, col] < 0.01){
            wilcox_test[item, col] <- paste0(wilcox_test[item, col], "***")
        }else if (wilcox_test[item, col] < 0.0260){
            wilcox_test[item, col] <- paste0(wilcox_test[item, col], "**")
        }else if (wilcox_test[item, col] < 0.05){
            wilcox_test[item, col] <- paste0(wilcox_test[item, col], "*")
        }
    }
}
```

# Create Other Variable Chart
```{r message=FALSE, warning=FALSE, fig.width=10}
otherVariablesFigure <- ggarrange(locationFig, sexFig, ageFig, 
                    labels = c("A", "B", "C"), nrow = 2, ncol = 2)+
                    bgcolor("white")
otherVariablesFigure

ggsave("Plots/1pct/otherVariablesFigure.jpeg")
```
